server/nuxeo-launcher/src/main/java/org/nuxeo/launcher/NuxeoLauncher.java

```diff
@@ -1814,131 +1563,232 @@ protected void setConfigProperties() throws ConfigurationException, IOException,
+            // for errors in doStart
+            hook.close();
+            throw e;
+        }
+        if (!quiet) {
+            log.info("Go to {}", this::getURL);
+        }
     }
 
-    protected void removeShutdownHook() {
+    /**
+     * Starts the server in background.
+     *
+     * @return the nuxeo process
+     * @throws NuxeoLauncherException if an error occurred
+     */
+    protected Process doStart(boolean logProcessOutput) {
         try {
-            Runtime.getRuntime().removeShutdownHook(shutdownHook);
-            log.debug("Removed shutdown hook");
+            configure();
+            configurationGenerator.verifyInstallation();
+
+            log.debug("Check if install in progress...");
+            int tries = 0;
+            while (configurationGenerator.isInstallInProgress()) {
+                tries++;
+                if (!getConnectBroker().executePending(configurationGenerator.getInstallFile(), true, true,
+                        ignoreMissing) || tries > 9) {
+                    throw new NuxeoLauncherException(String.format(
+                            "Start interrupted due to failure on pending actions. You can resume with a new start"
+                                    + " or you can restore the file '%s', optionally using the '--%s' option.",
+                            configurationGenerator.getInstallFile().getName(), OPTION_IGNORE_MISSING), EXIT_CODE_ERROR);
+                }
+                // reload configuration
+                configurationGenerator = new ConfigurationGenerator(quiet, debug);
+                configurationGenerator.init();
+                configure();
+                configurationGenerator.verifyInstallation();
+            }
+            return start(logProcessOutput);
+        } catch (ConfigurationException e) {
+            throw new NuxeoLauncherException("Could not run configuration: " + e.getMessage(), EXIT_CODE_NOT_CONFIGURED,
+                    e);
+        } catch (IOException e) {
+            throw new NuxeoLauncherException("Could not start process: " + e.getMessage(), EXIT_CODE_ERROR, e);
         } catch (IllegalStateException e) {
-            // the virtual machine is already in the process of shutting down
+            // in strict mode assume program is not configured because of http port binding conflict for exit value
+            throw new NuxeoLauncherException("Could not start process: " + e.getMessage(),
+                    strict ? EXIT_CODE_NOT_CONFIGURED : EXIT_CODE_ERROR, e);
         }
     }
 
     /**
-     * @return true if Nuxeo is ready
+     * Do not directly call this method without a call to {@link #checkNoRunningServer()}
+     *
+     * @see #doStart()
+     * @throws IOException in case of issue with process.
      */
-    protected boolean waitForEffectiveStart() throws InterruptedException {
-        long startTime = new Date().getTime();
+    protected Process start(boolean logProcessOutput) throws IOException {
```

Comment:
The reordering of the `start` / `waitForEffectiveStart` methods make the diff hard to read.

Reply:
Sorry about that, it was a pain to scroll 1K lines to jump between doStart and start. Do you want me to isolate reordering into one commit?

Reply:
Ideally it would be better but you shouldn't spend time on it of others have validated your changes.